### ISSUE TITLE
Send-WinSCPItem -Remove Switch

### ISSUE THREAD

**Issue by GHRoss** [dotps1/WinSCP#41]
The Send-WinSCPItem -Remove switch will remove not only the source file, but the folder that it is contained in.

Example:

`$filetransfer = Send-WinSCPItem -Path $transferitems -Destination $site.destPath -Remove`

This is moving a file from a location on the local E:\ drive to the /testing folder on the SFTP site then deleting only the file after. However as mentioned, it deletes the folder it's contained in too.

`$transferitems` looks like this: `$transferitems = Get-Item -Path $site.sourcePath`

**Comment by dotps1** [comment]
is $site.sourcePath a file or folder?

**Comment by GHRoss** [comment]
It's a folder.

________________________________

**Comment by dotps1** [comment]
well that would make sense then your saying get a folder and send it, then remove it when your finished.

**Comment by GHRoss** [comment]
I would agree, however it's not uploading the folder itself, only the files that are in the folder.

**Comment by dotps1** [comment]
Does your destination path have a trailing slash?

**Comment by GHRoss** [comment]
Good question, I think I tried it with and without the trailing slash. Would the fact that we get the files to upload with Get-Item make a difference?



On Wed, Jul 6, 2016 at 9:57 PM +0100, "Thomas Malkewitz" <notifications@github.com<mailto:notifications@github.com>> wrote:

Does your destination path have a trailing slash?

Sent from my iPhone

> On Jul 6, 2016, at 4:36 PM, GHRoss <notifications@github.com> wrote:
>
> I would agree, however it's not uploading the folder itself, only the files that are in the folder.
>
> -
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

-

**Comment by dotps1** [comment]
I will take a look at this but im pretty sure the way it works is if there is a '/' at the end, it thinks that is a the folder you would like them in, and then it will put all the files in that directory, and if there is NOT a '/' at the end, it will transfer the entire folder.

thanks for reporting.

**Comment by GHRoss** [comment]
We managed to make this work actually, I'll post how we did it tomorrow.



On Thu, Jul 7, 2016 at 6:36 PM +0100, "Thomas Malkewitz" <notifications@github.com<mailto:notifications@github.com>> wrote:


I will take a look at this but im pretty sure the way it works is if there is a '/' at the end, it thinks that is a the folder you would like them in, and then it will put all the files in that directory, and if there is NOT a '/' at the end, it will transfer the entire folder.

thanks for reporting.

-
